gigascience/gigadb-website
# pager text input for samples not appearing

# Pull request for issue: #1757

This is a pull request for the following functionalities:

* Add pagination to samples table in dataset view
![image](https://github.com/gigascience/gigadb-website/assets/143437854/9e544f13-96d5-4257-be98-622fff8f2e45)


## How to test?

Note: there's a bug I didn't fix, see in the next sections

* Navigate to a dataset with samples e.g. http://gigadb.gigasciencejournal.com/dataset/view/id/100035 and verify pagination widget is present
* enter any number in the input and press enter and verify that you navigate to the expected URL
* enter any number out of bounds, e.g. -4, 6 and verify that opens the min and max pages respectively
* leave input empty and press enter, verify that by default the first page is opened
* navigate to dataset with less than 5 samples and verify that pagination is not displayed,e .g. http://gigadb.gigasciencejournal.com/dataset/100006

## How have functionalities been implemented?

* Added pagination widget similarly as the one used for the files table
* Refactored and extended script related to pagination to be able to handle both paginations

## Any issues with implementation?

It looks like that for the http://gigadb.gigasciencejournal.com/dataset/view/id/100035 there is some kind of bug when calculating or retrieving the number of samples to display based on the current page and the number of results per page:

* The default shows 5 samples out of 18
![image](https://github.com/gigascience/gigadb-website/assets/143437854/72147d2f-c3cf-4a5c-9df6-d276d7b08778)
* Setting the items per page to 10 still shows 5 samples, rather than 10. However the pagination seems correct.
![image](https://github.com/gigascience/gigadb-website/assets/143437854/19ff979c-277d-464d-b35d-8bd9a17fe1f7)
* Going to any of the pages seems to show the same 5 samples, seemingly suggesting there are 5 samples and not 18
![image](https://github.com/gigascience/gigadb-website/assets/143437854/9f9a85fe-0cc4-4a5e-9c64-d8fee71d81c8)
* In the file `data/dev/sample.csv` I see 18 samples whose `name` seems to match the ones shown in the UI, so assuming this is correct there are in fact 18 samples
* If I inspect the php objects:
  * `var_dump($sampleDataProvider->getData())` prints an array with 5 items (5 samples). Each sample has 18 key-value pairs. The fact that there are 18 key-value pairs I assume is coincidental and unrelated to the total number of samples
  * `var_dump($model->samples)`, this seems to be a direct representation of a db entity, and it's an array of 18 elements (the number of samples), seems to point out to the correct number of samples
  * This data seems to be all defined in `protected/components/FormattedDatasetSamples.php`  `function getDataProvider`

Is there a bug in that function that results in returning an incorrect number of samples? Why does the `$sampleDataProvider->getData()` always return (the same) 5 samples, even if it's supposed to be 18 total?

## Any changes to automated tests?

--
